clientupdate/clientupdate.go

```diff
@@ -415,17 +415,25 @@ func (up *Updater) updateDebLike() error {
 		// we're not updating them:
 		"-o", "APT::Get::List-Cleanup=0",
 	)
-	cmd.Stdout = up.Stdout
-	cmd.Stderr = up.Stderr
-	if err := cmd.Run(); err != nil {
-		return err
+	if out, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("apt-get update failed: %w; output:\n%s", err, out)
 	}
 
-	cmd = exec.Command("apt-get", "install", "--yes", "--allow-downgrades", "tailscale="+ver)
-	cmd.Stdout = up.Stdout
-	cmd.Stderr = up.Stderr
-	if err := cmd.Run(); err != nil {
-		return err
+	for i := 0; i < 2; i++ {
+		out, err := exec.Command("apt-get", "install", "--yes", "--allow-downgrades", "tailscale="+ver).CombinedOutput()
+		if err != nil {
+			if !bytes.Contains(out, []byte(`dpkg was interrupted`)) {
+				return fmt.Errorf("apt-get install failed: %w; output:\n%s", err, out)
+			}
+			up.Logf("apt-get install failed: %s; output:\n%s", err, out)
+			up.Logf("running dpkg --configure tailscale")
+			out, err = exec.Command("dpkg", "--configure", "tailscale").CombinedOutput()
```

Comment:
are there scenarios where `dpkg --configure` will be in interactive mode?  Do we need to handle that here?  It seems the only options are the various `--force-conf*` flags which should force noninteractive mode, but I'm not sure that we have a safe option to always apply there.

Reply:
Good point! I think `--force-confdef` and `--force-downgrade` should be safe here.
```
downgrade(*): Install a package, even if newer version of it is already installed.

confdef: If a conffile has been modified and the version in the package did change, always choose the default action without prompting. If
           there is no default action it will stop to ask the user unless --force-confnew or --force-confold is also been given, in which case it will
           use that to decide the final action.
```
Both `--force-confnew` and `--force-confold` seem situational and I'd rather the user manually resolve those conflicts.

Reply:
yeah, that should be safe.